frontend/src/metabase/components/OnClickOutsideWrapper.jsx

```diff
@@ -19,10 +19,22 @@ export default class OnClickOutsideWrapper extends ClickOutComponent {
             // NOTE: this actually doesn't seem to be working correctly for popovers since PopoverBody creates a stacking context
             ReactDOM.findDOMNode(this).parentNode.style.zIndex = popoverStack.length + 2; // HACK: add 2 to ensure it's in front of main and nav elements
         }, 10);
+
+        // we unfortunately need to have this function here to access the proper context
+        const handleKeyPress = (event) => {
+            const escKey = 27; // the char code for the esc key
+            if (event.keyCode === escKey) {
+                event.preventDefault();
+                this.props.handleDismissal(event)
+            }
+        }
+
+        document.addEventListener('keydown', handleKeyPress)
     }
 
     componentWillUnmount() {
         super.componentWillUnmount();
+        document.removeEventListener('keydown', false)
```

Comment:
You need to pass exactly the same arguments you passed to `addEventListener` for it to be removed. I'd move `handleKeyPress` to be an instance method (just remove `const` and put it next to `componentDidMount` so "this" will be bound to the instance) then do `document.addEventListener('keydown', this.handleKeyPress, false)` and `document.removeEventListener('keydown', this.handleKeyPress, false)`, respectively.